compiler/backend/structuredLangScript.sml

```diff
@@ -0,0 +1,53 @@
+open preamble jsonTheory backend_commonTheory;
+
+val _ = new_theory"structuredLang";
+
+
+(* structuredLang is an intermediate language between presLang and json. The
+* purpose of strucutredLang is to force each presLang expression into a unified
+* form, that is easily expressed by uniform json object that are either
+*   a) Objects representing a tuple (Tuple)
+*   b) Objects representing a constructor with many arguments or (Item)
+*   c) Arrays (List)
+*   *)
+val _ = Datatype`
+  sExp =
+    | Tuple (sExp list)
+    | Item (tra option) string (sExp list)
+    | List (sExp list)`;
+
+(* structured_to_json *)
+val num_to_json_def = Define`
+  num_to_json n = String (num_to_str n)`;
+
+val trace_to_json_def = Define`
```

Comment:
Just for consistency, and less WTF-factor. Since every other part of the tree has a "name" property.